### PR TITLE
Create IDE project config documentation

### PR DESCRIPTION
**Pull request by jwitz**
Resolves https://github.com/astronomer/docs/issues/1837

### CODE REVIEW

File: astro/cloud-ide/configure-project-environment.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+1. In the Cloud UI, select a Workspace and then select **Cloud IDE**.
+
+2. Select a project.
+
+3. On the **Pipelines** page click a pipeline name to open the pipeline editor.
+
+4. In a SQL or Warehouse SQL cell, click **Select connection** and select the connection you want to use to store the results of the cell. If you are configuring a Warehouse SQL cell, additionally configure the **Output Table** where you want to permanently store the results of the cell query. 
+
+You can then use jinja templating to call your configured database connection from your SQL cell queries. For example:
```
Comment (jwitz) [inline astro/cloud-ide/configure-project-environment.md:91]: @jlaneve Can you help fill out the details here? How do users actually engage with their connection in SQL cells?

File: astro/cloud-ide/configure-project-environment.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+   
+6. Optional. Click **Test connection** to check that the you correctly configured the connection. 
+
+7. Click **Create Connection**.
+
+The connection appears in the **Connections** table. To edit the connection, click **Edit** in the **Connections** table.
+
+### Use connections in SQL cells
+
+Currently, Airflow connections configured through the Astro Cloud IDE can be used only to configure external databases for SQL and Warehouse SQL cells. To specify a connection in one of these cell types:
```
Comment (jwitz) [inline astro/cloud-ide/configure-project-environment.md:81]: I made this assumption, but is there any way to configure connections for Python cells?

File: astro/cloud-ide/configure-project-environment.md [hunk 3]
```diff
@@ -0,0 +1,105 @@
+---
+sidebar_label: Configure your project environment
+title: Configure your Astro Cloud IDE project environment
+id: configure-project-environment
+description: Learn how to configure environment variables, connections, and dependencies for use in your Astro cloud IDE pipelines.
+---
+
+Configure your Astro Cloud IDE project environment, including Airflow dependencies, variables, and connections, directly in the cloud UI. You can use your environment configurations in any pipeline within your project.
```
Comment (kwcanuck) [inline astro/cloud-ide/configure-project-environment.md:8]:
```suggestion
Configure your Astro Cloud IDE project environment, including Airflow dependencies, variables, and connections, in the Cloud UI. You can use your environment configurations in any pipeline within your project.
```

File: astro/cloud-ide/configure-project-environment.md [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+
+Environment configuration in the Astro Cloud IDE is similar Astro project and Airflow configuration, but includes more options for specifying values that minimize the time you spend configuring through text files.
```
Comment (kwcanuck) [inline astro/cloud-ide/configure-project-environment.md:10]:
```suggestion
Environment configuration in the Astro Cloud IDE is similar to Astro project and Airflow configuration, but more options are provided to minimize the time you spend configuring text files. 
```

File: astro/cloud-ide/configure-project-environment.md [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+
+2. Select a project.
+
+3. Click **Connections**, then click **+ Connection** to add a new environment variable.
+
+4. Choose a connection type.
+
+5. Configure the connection. 
+   
+6. Optional. Click **Test connection** to check that the you correctly configured the connection.
```
Comment (kwcanuck) [inline astro/cloud-ide/configure-project-environment.md:73]:
```suggestion
6. Optional. Click **Test connection** to check that you configured the connection correctly. 
```

File: astro/cloud-ide/configure-project-environment.md [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+description: Learn how to configure environment variables, connections, and dependencies for use in your Astro cloud IDE pipelines.
+---
+
+Configure your Astro Cloud IDE project environment, including Airflow dependencies, variables, and connections, directly in the cloud UI. You can use your environment configurations in any pipeline within your project.
+
+Environment configuration in the Astro Cloud IDE is similar Astro project and Airflow configuration, but includes more options for specifying values that minimize the time you spend configuring through text files. 
+
+## Limitations
+
+- You can't currently set OS-level dependencies through the Astro Cloud IDE.
```
Comment (kwcanuck) [inline astro/cloud-ide/configure-project-environment.md:14]:
```suggestion
- You can't set OS-level dependencies with the Astro Cloud IDE.
```

File: astro/cloud-ide/configure-project-environment.md [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+- Only a subset of Airflow connections can be configured from the Astro Cloud IDE.
```
Comment (kwcanuck) [inline astro/cloud-ide/configure-project-environment.md:15]:
```suggestion
- Only a subset of Airflow connections can be configured with the Astro Cloud IDE.
```

File: astro/cloud-ide/configure-project-environment.md [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+- Airflow connections configured through the Astro Cloud IDE can be used only to configure external databases for SQL and Warehouse SQL cells.
```
Comment (kwcanuck) [inline astro/cloud-ide/configure-project-environment.md:16]:
```suggestion
- Airflow connections configured with the Astro Cloud IDE can only be used to configure external databases for SQL and Warehouse SQL cells.
```

File: astro/cloud-ide/configure-project-environment.md [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+- Environment configurations are not saved when you export a data pipeline as a DAG. To run your data pipeline outside of the Astro Cloud IDE, you must reconfigure these values in the Airflow environment where you exported your data pipeline.
+
+## Add Python package requirements
+
+Setting Python package requirements in the Astro Cloud IDE is equivalent to setting them in the Astro project `requirements.txt` file or running `pip install <package>`.
+
+1. In the Cloud UI, select a Workspace and then select Cloud IDE.
```
Comment (kwcanuck) [inline astro/cloud-ide/configure-project-environment.md:23]:
```suggestion
1. In the Cloud UI, select a Workspace and then select **Cloud IDE**.
```

File: astro/cloud-ide/configure-project-environment.md [hunk 10]
```diff
@@ -0,0 +1,105 @@
...
+
+2. Select a project.
+
+3. Click **Requirements**, then click **+ Requirement** to add a new requirement.
```
Comment (kwcanuck) [inline astro/cloud-ide/configure-project-environment.md:27]:
```suggestion
3. Click the **Requirements** tab, then click **Requirement** to add a new requirement.
```

File: astro/cloud-ide/configure-project-environment.md [hunk 11]
```diff
@@ -0,0 +1,105 @@
...
+   
+4. Enter the name and version for the Python package. 
+
+5. Click **Add**.
+
+The package requirement appears in the **Requirements** table. To edit the Python package dependency, click **Edit** in the **Requirements** table.
```
Comment (kwcanuck) [inline astro/cloud-ide/configure-project-environment.md:33]:
```suggestion
The package requirement appears in the **Requirements** list. To edit the Python package dependency, click **Edit** in the **Requirements** list.
```

_The package requirement appears in the **Requirements** list._  - if this is obvious application behavior, you don't need to describe it. 

Your final step could be:

6. Optional. To edit the Python package dependency, click **Edit** in the **Requirements** list.

File: astro/cloud-ide/configure-project-environment.md [hunk 12]
```diff
@@ -0,0 +1,105 @@
...
+
+## Configure environment variables
+
+You can configure environment variables in the Astro Cloud IDE in the same way that you can for Astro Deployments. Similarly, Astro Cloud IDE environment variables are stored and retrieved in the same way as Deployment environment variables. See [How environment variables are stored on Astro](environment-variables.md#how-environment-variables-are-stored-on-astro).
```
Comment (kwcanuck) [inline astro/cloud-ide/configure-project-environment.md:37]:
```suggestion
The process for configuring environment variables in the Astro Cloud IDE and in Deployments is very similar. Also, Astro Cloud IDE environment variables are stored and retrieved in the same way as Deployment environment variables. See [How environment variables are stored on Astro](environment-variables.md#how-environment-variables-are-stored-on-astro).
```

File: astro/cloud-ide/configure-project-environment.md [hunk 13]
```diff
@@ -0,0 +1,105 @@
...
+
+Additionally, the Astro Cloud IDE lets you specify whether an environment variable should be used as either an Airflow variable or a system level environment variable.
```
Comment (kwcanuck) [inline astro/cloud-ide/configure-project-environment.md:39]:
```suggestion
In the Astro Cloud IDE, you can specify whether an environment variable should be used as an Airflow variable or a system level environment variable.
```

File: astro/cloud-ide/configure-project-environment.md [hunk 14]
```diff
@@ -0,0 +1,105 @@
...
+
+1. In the Cloud UI, select a Workspace and then select Cloud IDE.
+
+2. Select a project.
+
+3. Click **Variables**, then click **+ Variable** to add a new environment variable.
```
Comment (kwcanuck) [inline astro/cloud-ide/configure-project-environment.md:45]:
```suggestion
3. Click the **Variables** tab, and then click **Variable**.
```

File: astro/cloud-ide/configure-project-environment.md [hunk 15]
```diff
@@ -0,0 +1,105 @@
...
+
+4. Configure the following values:
```
Comment (kwcanuck) [inline astro/cloud-ide/configure-project-environment.md:47]:
```suggestion
4. Complete the following fields:
```

File: astro/cloud-ide/configure-project-environment.md [hunk 16]
```diff
@@ -0,0 +1,105 @@
...
+
+    - **Type**: Select the purpose of the environment variable. Select **Environment** if you want to configure a system-level environment variable for your runtime environment, such as `AIRFLOW__CORE__DEFAULT_TASK_EXECUTION_TIMEOUT`. Select **Airflow** if you want to call the variable key from a Python cell using `Variable.get('<variable-name>')`. If you select **Airflow**, do not specify your environment variable key with `AIRFLOW_VAR_`. 
+    - **Key**: They key for your environment variable.
+    - **Value**: The value for your environment variable.
+
+5. Optional. Check **Mask Value** to make the variable value secret.
+
+6. Click **Create Variable**.
+
+The environment variable appears in the **Variables** table. To edit the environment variable, click **Edit** in the **Variables** table.
```
Comment (kwcanuck) [inline astro/cloud-ide/configure-project-environment.md:57]:
```suggestion
The environment variable appears in the **Variables** list. To edit the environment variable, click **Edit** in the **Variables** list.
```
The environment variable appears in the **Variables** list. - if this is obvious application behavior, you don't need to describe it.

Your final step could be:

Optional. To edit the environment variable, click **Edit** in the **Variables** list.

File: astro/cloud-ide/configure-project-environment.md [hunk 17]
```diff
@@ -0,0 +1,105 @@
...
+
+## Configure Airflow connections
+
+You can configure Airflow connections in the Astro Cloud IDE in the same way that you can in the [Airflow UI](https://docs.astronomer.io/learn/connections). You can then reference the connection in your Python cells as code or in SQL cells as a configuration
+
+1. In the Cloud UI, select a Workspace and then select Cloud IDE.
```
Comment (kwcanuck) [inline astro/cloud-ide/configure-project-environment.md:63]:
```suggestion
1. In the Cloud UI, select a Workspace and then select **Cloud IDE**.
```

File: astro/cloud-ide/configure-project-environment.md [hunk 18]
```diff
@@ -0,0 +1,105 @@
...
+
+2. Select a project.
+
+3. Click **Connections**, then click **+ Connection** to add a new environment variable.
+
+4. Choose a connection type.
```
Comment (kwcanuck) [inline astro/cloud-ide/configure-project-environment.md:69]:
```suggestion
4. Select a connection type.
```

File: astro/cloud-ide/configure-project-environment.md [hunk 19]
```diff
@@ -0,0 +1,105 @@
...
+
+## Configure Airflow connections
+
+You can configure Airflow connections in the Astro Cloud IDE in the same way that you can in the [Airflow UI](https://docs.astronomer.io/learn/connections). You can then reference the connection in your Python cells as code or in SQL cells as a configuration
+
+1. In the Cloud UI, select a Workspace and then select Cloud IDE.
+
+2. Select a project.
+
+3. Click **Connections**, then click **+ Connection** to add a new environment variable.
```
Comment (kwcanuck) [inline astro/cloud-ide/configure-project-environment.md:67]:
```suggestion
3. Click the **Connections** tab, and then click **Connection**.
```

File: astro/cloud-ide/configure-project-environment.md [hunk 20]
```diff
@@ -0,0 +1,105 @@
...
+
+1. In the Cloud UI, select a Workspace and then select Cloud IDE.
+
+2. Select a project.
+
+3. Click **Variables**, then click **+ Variable** to add a new environment variable.
+
+4. Configure the following values:
+
+    - **Type**: Select the purpose of the environment variable. Select **Environment** if you want to configure a system-level environment variable for your runtime environment, such as `AIRFLOW__CORE__DEFAULT_TASK_EXECUTION_TIMEOUT`. Select **Airflow** if you want to call the variable key from a Python cell using `Variable.get('<variable-name>')`. If you select **Airflow**, do not specify your environment variable key with `AIRFLOW_VAR_`.
```
Comment (jlaneve) [inline astro/cloud-ide/configure-project-environment.md:49]: do you think we should break the types out into sub-bullets? it's a bit hard to read right now with regular text, bold text, and code in a few long lines

File: astro/cloud-ide/configure-project-environment.md [hunk 21]
```diff
@@ -0,0 +1,105 @@
...
+
+4. Choose a connection type.
+
+5. Configure the connection. 
+   
+6. Optional. Click **Test connection** to check that the you correctly configured the connection. 
+
+7. Click **Create Connection**.
+
+The connection appears in the **Connections** table. To edit the connection, click **Edit** in the **Connections** table.
```
Comment (kwcanuck) [inline astro/cloud-ide/configure-project-environment.md:77]:
```suggestion
The connection appears in the **Connections** list. To edit the connection, click **Edit** in the **Connections** list.
```
_The connection appears in the **Connections** list._ - if this is obvious application behavior, you don't need to describe it.

Your final step could be:

Optional. To edit the connection settings, click **Edit** in the **Connections** list.

File: astro/cloud-ide/configure-project-environment.md (hunk 2 above)
Comment (kwcanuck) [inline astro/cloud-ide/configure-project-environment.md:81]:
```suggestion
Airflow connections configured through the Astro Cloud IDE can be used only to configure external databases for SQL and Warehouse SQL cells.
```

File: astro/cloud-ide/configure-project-environment.md (hunk 7 above)
Comment (jlaneve) [inline astro/cloud-ide/configure-project-environment.md:15]: not sure if this is worth clarifying, but there's a `Generic` connection type that lets you pick a connection type for connections we haven't explicitly built out the form yet

File: astro/cloud-ide/configure-project-environment.md [hunk 22]
```diff
@@ -0,0 +1,105 @@
...
+
+### Use connections in SQL cells
+
+Currently, Airflow connections configured through the Astro Cloud IDE can be used only to configure external databases for SQL and Warehouse SQL cells. To specify a connection in one of these cell types:
+
+1. In the Cloud UI, select a Workspace and then select **Cloud IDE**.
+
+2. Select a project.
+
+3. On the **Pipelines** page click a pipeline name to open the pipeline editor.
```
Comment (kwcanuck) [inline astro/cloud-ide/configure-project-environment.md:87]:
```suggestion
3. Click the **Pipelines** tab, and then click a pipeline name to open the pipeline editor.
```

File: astro/cloud-ide/configure-project-environment.md (hunk 2 above)
Comment (jlaneve) [reply astro/cloud-ide/configure-project-environment.md:81]: kind of - your connections are still available to you in python cells, the use case is just a bit different. for example, you could do something like `S3Hook.check_for_bucket(...)` using an S3 connection in python

File: astro/cloud-ide/configure-project-environment.md (hunk 1 above)
Comment (kwcanuck) [inline astro/cloud-ide/configure-project-environment.md:91]:
```suggestion
5. Optional. Use jinja templating to call your configured database connection from your SQL cell queries. For example:
```

File: astro/cloud-ide/configure-project-environment.md [hunk 23]
```diff
@@ -0,0 +1,105 @@
...
+
+4. In a SQL or Warehouse SQL cell, click **Select connection** and select the connection you want to use to store the results of the cell. If you are configuring a Warehouse SQL cell, additionally configure the **Output Table** where you want to permanently store the results of the cell query. 
+
+You can then use jinja templating to call your configured database connection from your SQL cell queries. For example:
+
+```sql
+SELECT * FROM {{<connection-id>}};
```
Comment (jlaneve) [inline astro/cloud-ide/configure-project-environment.md:94]: selecting from a connection doesn't really make sense, the sql just runs on the connection you specify in the dropdown. so if you had a table in your connection called `table_name`, you would do something like:

```suggestion
SELECT * FROM table_name;
```

File: astro/cloud-ide/configure-project-environment.md [hunk 24]
```diff
@@ -0,0 +1,105 @@
...
+```
+
+## View environment configurations from the pipeline editor
+
+Environment configurations apply to all pipelines in a given project. To view your configurations from the pipeline editor, click **Environment**. The **Use in your pipeline…** menu shows all configurations which apply to your current pipeline. You can also use this menu to add, delete, or modify environment configurations.
```
Comment (kwcanuck) [inline astro/cloud-ide/configure-project-environment.md:99]:
```suggestion
Environment configurations apply to all pipelines in a project. To view your configurations in the pipeline editor, click **Environment**. The **Use in your pipeline** pane shows all configurations that apply to your current pipeline. You can add, delete, or modify environment configurations in the pane.
```
image

File: astro/cloud-ide/configure-project-environment.md [hunk 25]
```diff
@@ -0,0 +1,105 @@
...
+
+:::info
+
+Because environment configurations exist at the project level, modifying them from your pipeline editor will update the configurations for all pipelines in your project. To run a data pipeline with different environment configurations from its existing IDE project, you must recreate it in a new IDE project.
```
Comment (kwcanuck) [inline astro/cloud-ide/configure-project-environment.md:103]:
```suggestion
Environment configurations exist at the project level. Modifying them in your pipeline editor updates the configurations for all pipelines in your project. To run a data pipeline with different environment configurations from its existing IDE project, you must recreate it in a new IDE project. 
```

**Review by kwcanuck — APPROVED** [review]

File: astro/cloud-ide/configure-project-environment.md (hunk 10 above)
Comment (jwitz) [reply astro/cloud-ide/configure-project-environment.md:27]: Is there anything else we can say about the **Requirement** button to distinguish it from **Requirements**? I don't want users to get frustrated over that minute difference.

File: astro/cloud-ide/configure-project-environment.md [hunk 26]
```diff
@@ -0,0 +1,105 @@
...
+
+2. Select a project.
+
+3. Click **Variables**, then click **+ Variable** to add a new environment variable.
+
+4. Configure the following values:
+
+    - **Type**: Select the purpose of the environment variable. Select **Environment** if you want to configure a system-level environment variable for your runtime environment, such as `AIRFLOW__CORE__DEFAULT_TASK_EXECUTION_TIMEOUT`. Select **Airflow** if you want to call the variable key from a Python cell using `Variable.get('<variable-name>')`. If you select **Airflow**, do not specify your environment variable key with `AIRFLOW_VAR_`. 
+    - **Key**: They key for your environment variable.
+    - **Value**: The value for your environment variable.
```
Comment (jwitz) [inline astro/cloud-ide/configure-project-environment.md:51]:
```suggestion
    - **Type**: Select the purpose of the environment variable. 
        - Select **Environment** if you want to configure a system-level environment variable for your runtime environment, such as `AIRFLOW__CORE__DEFAULT_TASK_EXECUTION_TIMEOUT`. 
        - Select **Airflow** if you want to call the variable key from a Python cell using `Variable.get('<variable-name>')`. If you select **Airflow**, do not specify your environment variable key with `AIRFLOW_VAR_`. 
    - **Key**: They key for your environment variable.
    - **Value**: The value for your environment variable.
```

File: astro/cloud-ide/configure-project-environment.md [hunk 27]
```diff
@@ -0,0 +1,106 @@
...
+5. Configure the connection. 
+   
+6. Optional. Click **Test connection** to check that you configured the connection correctly. Note that you cannot test generic connections.
+
+7. Click **Create Connection**.
+
+The connection appears in the **Connections** list. To edit the connection, click **Edit** in the **Connections** list.
+
+### Use connections in SQL cells
+
```
Comment (jwitz) [inline astro/cloud-ide/configure-project-environment.md:82]:
```suggestion
### Use connections in cells
```

File: astro/cloud-ide/configure-project-environment.md [hunk 28]
```diff
@@ -0,0 +1,106 @@
...
+
```
Comment (jwitz) [inline astro/cloud-ide/configure-project-environment.md:83]:
```suggestion

To use a connection in a Python cell, pass the connection ID to any function that accepts an Airflow connection as an argument, such as a [hook](https://docs.astronomer.io/learn/what-is-a-hook).

To use a connection in a SQL or Warehouse SQL cell:
```

File: astro/cloud-ide/configure-project-environment.md (hunk 10 above)
Comment (kwcanuck) [reply astro/cloud-ide/configure-project-environment.md:27]: @jwitz typically, you don't include symbols or punctuation (ellipsis, for example) when referencing UI controls. The _Google developer documentation style guide_  recommends using the `aria-label` attribute to determine the correct control name. Using that advice, here is the correct name for that control:

image

Unfortunately, using a control name that doesn't match the UI will likely confuse users. It is for this reason I didn't recommend using the `aria-label` attribute. If you're concerned that users will be unable to locate the control, provide a screen capture, or location information (which is also taboo).

File: astro/cloud-ide/configure-project-environment.md [hunk 29]
```diff
@@ -0,0 +1,126 @@
...
+### Call Airflow variables in cells 
+
+Call Airflow variable values in Python cells using `Variable.get('<variable-key>')`. For example:
+
+```python
+value = Variable.get('MY_KEY')
+print(value)
+```
+
+Call Airflow variables in SQL or Warehouse SQL cells using jinja templating. For example, if your environment variable stored the name of a column, you could run:
```
Comment (jwitz) [inline astro/cloud-ide/configure-project-environment.md:72]:
```suggestion
Call Airflow variables in SQL or Warehouse SQL cells using jinja templating. For example, if your environment variable value is the name of a column in `Mytable`, you could run:
```

File: astro/cloud-ide/configure-project-environment.md [hunk 30]
```diff
@@ -0,0 +1,126 @@
...
+
+```sql
+SELECT {{ var.value.MY_KEY }} FROM Mytable;
```
Comment (jwitz) [inline astro/cloud-ide/configure-project-environment.md:75]:
```suggestion
SELECT {{ var.value.MY_KEY }} FROM mytable;
```

**Pull request merged**